kernel/thread.c

```diff
@@ -590,6 +590,25 @@ void _k_thread_single_abort(struct k_thread *thread)
 }
 
 #ifdef CONFIG_MULTITHREADING
+#ifdef CONFIG_USERSPACE
+extern char __object_access_start[];
+extern char __object_access_end[];
+
+static void grant_static_access(void)
+{
+	struct _k_object_assignment *pos;
+
+	for (pos = (struct _k_object_assignment *)__object_access_start;
+	     pos < (struct _k_object_assignment *)__object_access_end;
```

Comment:
i think it should be <= in condition statement, otherwise we will not be processing the last object.

Reply:
no, definitely not
__object_access_end does not point to a struct _k_object_assignment!

Reply:
yeah i realized that and removed the comment later but it didnt reflect for some reason.